benchmarks/expression_problem/functional_f#/Program.fs

```diff
@@ -0,0 +1,48 @@
+﻿type Expr = Add of Expr * Expr
+          | Var of int
+          | Neg of Expr
+          | Mul of Expr * Expr
+          | Min of Expr * Expr
+
+let rec prettyPrint e =
+    match e with
+      | Add(expr1,expr2) -> sprintf "(%s+%s)" (prettyPrint expr1) (prettyPrint expr2)
+      | Var(num) -> sprintf "%i" num
+      | Neg(expr) -> sprintf "(-%s)" (prettyPrint expr)
+      | Mul(expr1,expr2) -> sprintf "(%s*%s)" (prettyPrint expr1) (prettyPrint expr2)
+      | Min(expr1,expr2) -> sprintf "(%s-%s)" (prettyPrint expr1) (prettyPrint expr2)
+
+let rec eval e =
+    match e with
+      | Add(expr1,expr2) -> (eval expr1) + (eval expr2)
+      | Var(num) -> num
+      | Neg(expr) -> -(eval expr)
+      | Mul(expr1,expr2) -> (eval expr1) * (eval expr2)
+      | Min(expr1,expr2) -> (eval expr1) - (eval expr2)
+
+let rand = System.Random(2)

```

Comment:
Burde denne måske ikke ligge globalt, da den introducerer state. Måske den bare skulle gives som input parameter?

Reply:
Ligger den ikke globalt?

Reply:
Når man giver random som argument får man e kopi og ikke en reference. Man kunne godt give den som parameter i stedet for, men så kommer det til at give et andet resultat end det de andre implementationer får.